react/components/ListsPage/Content.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { Component, Fragment, ReactNode } from 'react'
+
+import classNames from 'classnames'
+import { append, filter, map, tail } from 'ramda'
+import { Spinner } from 'vtex.styleguide'
+
+import ApolloClient from 'apollo-client'
+import { withApollo, WithApolloClient } from 'react-apollo'
+import { getListDetailed, getListsIdFromCookies, updateList } from '../../GraphqlClient'
+import ListItems from '../ListDetails/Content'
+import Footer from '../ListDetails/Footer'
+import Header from './Header'
+
+import wishlist from '../../wishList.css'
+
+interface ContentProps {
+  listId: string
+  lists?: any
+  onListCreated: (list: List) => void
+  onListUpdated: (list: List) => void
+  onListDeleted: () => void
+  client?: ApolloClient<any>
+}
+
+interface ContentState {
+  list?: any
+  selectedItems: any
+  isLoading?: boolean
+}
+
+class Content extends Component<ContentProps & WithApolloClient<any>, ContentState> {
+  public state: ContentState = {
+    isLoading: true,
+    selectedItems: [],
+  }
+  private isComponentMounted = false
+
+  public componentWillUnmount(): void {
+    this.isComponentMounted = false
+  }
+
+  public componentDidMount(): void {
+    this.isComponentMounted = true
+    this.fetchListDetails()
+  }
+
+  public componentDidUpdate(prevProps: any): void {
+    if (this.props !== prevProps) {
```

Comment:
shouldn't you just check for the `listId` here? checking for the props object will trigger this on every update